helpers/buildHierarchy.ts

```diff
@@ -0,0 +1,20 @@
+import { Department } from 'types'
+
+function buildHierarchy(list: Department[]) {
+	return list.reduce((acc, item) => {
+		const id = item.id
+		const parentId = item.parent?.id
+		if (!parentId) {
+			acc.push(item)
+		} else {
+			const parent = acc.find((dep) => dep.id === parentId)
+			if (parent) {
+				parent.children = parent.children.map((child) =>
+					child.id === id ? item : child
+				)
+			}
+		}
+		return acc
+	}, [])
```

Comment:
```suggestion
	}, [] as Department[])
```

Adding a case ensures that your `acc` argument gets type checked.